src/ImageSharp/Drawing/Paths/LinearLineSegment.cs

```diff
@@ -0,0 +1,83 @@
+﻿// <copyright file="LinearLineSegment.cs" company="James Jackson-South">
+// Copyright (c) James Jackson-South and contributors.
+// Licensed under the Apache License, Version 2.0.
+// </copyright>
+
+namespace ImageSharp.Drawing.Paths
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Collections.ObjectModel;
+    using System.Linq;
+    using System.Numerics;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// Represents a seriese of control points that will be joined by staight lines
+    /// </summary>
+    /// <seealso cref="ImageSharp.Drawing.Paths.ILineSegment" />
+    public class LinearLineSegment : ILineSegment
+    {
+        private Vector2[] points;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LinearLineSegment"/> class.
+        /// </summary>
+        /// <param name="points">The points.</param>
+        public LinearLineSegment(IEnumerable<PointF> points)
+            : this(points?.Select(x => x.ToVector2()).ToArray())
```

Comment:
these only happen at Contructor time and thus a one time hit, I think I had envisioned Shapes and this Paths and LineSegments being relatively long lived items and you wouldn't new a new one up each time if you wanted to apply it to many things, as the overhead required on things like the ComplexPolygon would be very high if newing them up all the time.